pkgs/development/beam-modules/mix-release.nix

```diff
@@ -8,15 +8,23 @@
 , enableDebugInfo ? false
 , mixEnv ? "prod"
 , compileFlags ? [ ]
-, mixDeps ? null
+  # mix fixed output derivation dependencies
+, mixFodDeps ? null
+  # mix dependencies generated by mix2nix
+  # this assumes each dependency is built by buildMix or buildRebar3
+  # each dependency needs to have a setup hook to add the lib path to $ERL_LIBS
+  # this is how mix will find dependencies
+, mixNixDeps ? { }
```

Comment:
I mentioned it in a comment on #123384, but I'm repeating myself here for visibility: I'm wondering if we can just use `buildInputs` instead of `mixNixDeps` (and `beamDeps` for `rebar3Relx`/`buildRebar3`). @happysalada you mentioned that `mixNixDeps` behaves a little differently though, do you mind explaining how it differs? In any case I think this is only a small thing and shouldn't be a blocker, we could change it across the whole of `beamPackages` in a later PR.

Reply:
Same, I'll repeat here to make sure.

In principle I like the idea of just using buildInputs.
The only obstacle I see is preserving the old FOD api. Just like in rebar, there needs to be a check that you only use one or the other (FOD or Nix deps). However if someone wants to use the FOD with buildInputs, he won't be able to.
I would suggest to deprecate the FOD api. That means changing the docs, and asking @minijackson to change his PR using mixRelease.
Let me know if I misunderstood something.

Reply:
One more thing regarding this. I guess, if people want to use the FOD, they should still be able to if we remove the parameter from the list. It would just mean that they need to do a couple more manual steps.
For me that decides in favor of removing FOD. We remove it from the parameters,, and if people still want to use it, they will need to override things and add a couple of manual steps.